### PR TITLE
14324 make pad required field

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity#14324

*Description of changes:*
- Made Pad text field required, changed validation
- Changes to Review and Confirm screen: hide section values unless changes have been made to form, some text changes as well

No changes to Location Type form looks like this on Review and Confirm:
image

Updated Review and Confirm screen for Pad number validation:
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: mhr_api/src/mhr_api/resources/v1/other_registrations.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def post_other_registration(mhr_number: str):
             return resource_utils.unauthorized_error_response(account_id)
         # Try to fetch summary registration by mhr number
         registration = MhrRegistration.find_summary_by_mhr_number(account_id, mhr_number)
-        if registration is None:
+        if not registration:
             return resource_utils.not_found_error_response('Manufactured Home registration', mhr_number)
 
         # Check if duplicate.
```
Comment (cameron-eyds) [inline mhr_api/src/mhr_api/resources/v1/other_registrations.py:90]: Is this related to the PR? 
Was this branch potentially cut from your previous API pr branch?

File: ppr-ui/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ppr-ui",
-  "version": "0.3.58",
+  "version": "0.3.59",
```
Comment (cameron-eyds) [inline ppr-ui/package.json]: Will need to `npm i` to update your plock as well. 
That being said, what version of Node is your system?
I noticed quite a large diff in the plock last time, which was probably due to the node version.

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue [hunk 3]
```diff
@@ -6,7 +6,8 @@
     </header>
 
     <div :class="{ 'border-error-left': !getStepValidation(MhrSectVal.LOCATION_VALID)}">
-      <section class="mx-6 pt-8" v-if="!getStepValidation(MhrSectVal.LOCATION_VALID)">
+      <section :class="{ 'pb-8': !(!!getMhrRegistrationLocation.locationType) && !hasAddress }" class="mx-6 pt-8"
+      v-if="!getStepValidation(MhrSectVal.LOCATION_VALID)">
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue]: This condition is a little hard to follow 😁 
Isn't `!(!!getMhrRegistrationLocation.locationType)` the same as `! getMhrRegistrationLocation.locationType` ?
Might be able to simplify that, if so.

Also missing an indent on your v-if. 
Usually, if you require a line break, you would break line for all the props, conditions etc. 
with the `>` on a new line as well, i'm sure there is some examples around.

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue [hunk 4]
```diff
@@ -16,7 +17,8 @@
         </span>
       </section>
 
-      <section class="py-6" id="review-home-location-section">
+      <section class="py-6" id="review-home-location-section"
+      v-if="(!!getMhrRegistrationLocation.locationType || hasAddress)">
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue]: Same as above, some minor formatting here

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue [hunk 5]
```diff
@@ -124,6 +124,11 @@ export default defineComponent({
         return [HomeLocationTypes.OTHER_STRATA, HomeLocationTypes.OTHER_TYPE]
           .includes(getMhrRegistrationLocation.value.otherType)
       }),
+      hasAddress: computed((): boolean => {
+        return getMhrRegistrationLocation.value.address.street ||
+        getMhrRegistrationLocation.value.address.streetAdditional ||
+        getMhrRegistrationLocation.value.address.city
+      }),
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue]: With longer accessors its always a good idea to optional change, in a computed value like this at least. 
if `address` is ever undefined, it will result in a type error, so an `?` after `address` would protect the conditions.

File: mhr_api/src/mhr_api/resources/v1/other_registrations.py (hunk 1 above)
Comment (owillborn) [reply mhr_api/src/mhr_api/resources/v1/other_registrations.py:90]: This was from another PR, got pulled in when I rebased BC it hasn't been merged yet

File: ppr-ui/package.json (hunk 2 above)
Comment (owillborn) [reply ppr-ui/package.json]: v16.17.0, will update plock

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue (hunk 3 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue]: Yes, you're right haha, will change that. Was looking for some examples on the line overlap formatting, will change as you describe

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue (hunk 4 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue]: will do

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue (hunk 5 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeLocationReview.vue]: missed that

File: mhr_api/src/mhr_api/resources/v1/other_registrations.py (hunk 1 above)
Comment (owillborn) [reply mhr_api/src/mhr_api/resources/v1/other_registrations.py:90]: I think

File: ppr-ui/package.json (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/package.json]: I believe we are using Node version `16.14.2` for specific reasons, fairly certain it was certain vulnerabilities in Node between a certain version set.
Could you roll back your Node Version before committing a new plock?
I can't recall the exact details but i imagine Omid might recall, there was a ticket awhile back to ensure this was the version we were on across the projects.

Comment (owillborn) [reply ppr-ui/package.json]: Will do, thanks for clarification

**Review by dimak1 — APPROVED** [review]
Apart from Cameron's comments, looks ok.

**Comment by cameron-eyds** [comment]
@owillborn Something is off with regards to the diff in the P Lock file still. For expediency, maybe revert your version bump and changes to the p lock until we can determine why.

**Review by cameron-eyds — APPROVED** [review]
Looks good, thanks. We will figure out the node version issue 👍
